processor/src/memory/mod.rs

```diff
@@ -209,17 +210,43 @@ impl Memory {
         }
     }
 
-    // TEST METHODS
-    // --------------------------------------------------------------------------------------------
-
-    /// Instantiates a new processor for testing purposes.
-    #[cfg(test)]
+    /// Reads a specific address of it's value.
+    /// Unlike read() that modifies the underlying map, get_value() only attempts to read
+    /// or return None when no value exists.
     pub fn get_value(&self, addr: u64) -> Option<Word> {
         match self.trace.get(&addr) {
             Some(addr_trace) => addr_trace.last().map(|(_, value)| *value),
             None => None,
         }
     }
+
+    /// Returns all the addresses and values stored in memory.
+    pub fn get_all_values(&self) -> Vec<(u64, Word)> {
+        let mut data: Vec<(u64, [Felt; 4])> = Vec::new();
+        for (&addr, addr_trace) in self.trace.iter() {
+            let value = match addr_trace.last() {
+                Some(t) => t.1,
+                None => INIT_MEM_VALUE,
+            };
+            data.push((addr, value));
+        }
+
+        data
+    }
```

Comment:
Should this be just a pass-through method to `get_values()` using range from 0 to `u64::MAX`?

Reply:
Oh that's a good point, I think there is a perf hit here but I'm not sure if it's signifcant (one needs to search leaf nodes, all just iterates through all the nodes).
I'll keep this simple for now until we see the need to improve.